packages/lodestar/src/network/gossip/validator.ts

```diff
@@ -42,90 +44,175 @@ export class GossipMessageValidator implements IGossipMessageValidator {
     this.logger = logger;
   }
 
+  async updateAttestationSeenCaches(aggregateAndProof: AggregateAndProof): Promise<void> {
+    await Promise.all([
+      this.db.aggregateAndProof.add(aggregateAndProof),
+      this.db.seenAttestationCache.addAggregateAndProof(aggregateAndProof),
+    ]);
+  }
+
   public isValidIncomingBlock = async (signedBlock: SignedBeaconBlock): Promise<ExtendedValidatorResult> => {
     try {
-      return await validateGossipBlock(this.config, this.chain, this.db, this.logger, signedBlock);
+      const blockJob = {
+        signedBlock: signedBlock,
+        trusted: false,
+        reprocess: false,
+      } as IBlockJob;
+      await validateGossipBlock(this.config, this.chain, this.db, this.logger, blockJob);
     } catch (e) {
       this.logger.error("Error while validating gossip block", e);
-      return ExtendedValidatorResult.ignore;
+      if (
+        e.code === BlockErrorCode.ERR_FUTURE_SLOT ||
+        e.code === BlockErrorCode.ERR_WOULD_REVERT_FINALIZED_SLOT ||
+        e.code === BlockErrorCode.ERR_REPEAT_PROPOSAL ||
+        e.code === BlockErrorCode.ERR_PARENT_UNKNOWN
```

Comment:
can you create two arrays one with ignorable errors and second with rejectable for every gossip topic and use them instead of this manual comparison